src/components/navigation/document-tab-content.tsx

```diff
@@ -0,0 +1,121 @@
+            return (
+              <Tab
+                className={`doc-tab ${leftTabSpec?.tab} ${section.type}`}
+                key={`section-${section.type}`}
+                data-test={section.dataTestHeader}
+              >
+                {sectionTitle}
+              </Tab>
+            );
+          })}
+        </TabList>
+        {leftTabSpec && leftTabSpec.sections.map((section) => {
+          const _handleDocumentStarClick = section.showStarsForUser(user)
+                ? this.handleDocumentStarClick
+                : undefined;
+          const _handleDocumentDeleteClick = section.showDeleteForUser(user)
+                ? this.handleDocumentDeleteClick
+                : undefined;
+          return (
+            <TabPanel key={`section-${section.type}`}>
+              <TabPanelDocumentsSection
+                key={section.type}
+                tab={leftTabSpec!.tab}
+                section={section}
+                stores={this.stores}
+                scale={kNavItemScale}
+                onNewDocumentClick={this.handleNewDocumentClick}
+                onDocumentClick={this.handleDocumentClick}
+                onDocumentDragStart={this.handleDocumentDragStart}
+                onDocumentStarClick={_handleDocumentStarClick}
+                onDocumentDeleteClick={_handleDocumentDeleteClick}
+              />
+            </TabPanel>
+          );
+        })}
+      </Tabs>
+    );
+  }
+
+  private getSectionTitle = (section: LeftTabSectionModelType, stores: IStores) => {
+    if (section.title === "%abbrevInvestigation%") {
+      const { unit, investigation } = stores;
+      const { abbrevTitle } = unit;
+      const prefix = abbrevTitle ? `${abbrevTitle}: ` : "";
+      return `${prefix}Investigation ${investigation.ordinal}`;
+    }
+    return section.title;
+  }
+
+  private handleNewDocumentClick = async (section: LeftTabSectionModelType) => {
+    const { appConfig: { defaultDocumentContent }, db, ui } = this.stores;
+    const { problemWorkspace } = ui;
+    const newDocument = section.type === ELeftTabSectionType.kPersonalDocuments
+                          ? await db.createPersonalDocument({ content: defaultDocumentContent })
+                          : await db.createLearningLogDocument();
+    if (newDocument) {
+      problemWorkspace.setAvailableDocument(newDocument);
+      ui.restoreDefaultNavExpansion();
+    }
+  }
+
+  private handleDocumentClick = (document: DocumentModelType) => {
+    const { appConfig, ui } = this.stores;
+    ui.rightNavDocumentSelected(appConfig, document);
+  }
+
+  private handleDocumentDragStart = (e: React.DragEvent<HTMLDivElement>, document: DocumentModelType) => {
+    e.dataTransfer.setData(DocumentDragKey, document.key);
+  }
+
+  private handleDocumentStarClick = (document: DocumentModelType) => {
+    const { user } = this.stores;
+    document && document.toggleUserStar(user.id);
```

Comment:
There are several places, like this one, where optional chaining could be used. I believe much of this code was copied from elsewhere and was originally written before optional chaining was available. It's not necessary to spend time tracking them all down but I think we should gradually make the change as we work with preexisting code.

Reply:
Agreed.  I fixed the instances that I found in this module.  We can fix others as we encounter them.